src/templates.ts

```diff
@@ -110,9 +111,35 @@ export function getTemplates(options: ModuleOptions, uiConfig: Record<string, an
   writeThemeTemplate(theme)
 
   async function generateSources() {
-    let sources = ''
+    if (!nuxt) {
+      return '@source "./ui";'
+    }
+
+    const sources: string[] = []
 
-    if (!!nuxt && !!resolve && options.experimental?.componentDetection) {
+    // Add layer sources
+    for (const layer of nuxt.options._layers || []) {
```

Comment:
You can use `getLayerDirectories()` from `@nuxt/kit` here: https://nuxt.com/docs/4.x/api/kit/layers#getlayerdirectories